From: Debian GNOME Maintainers
 <pkg-gnome-maintainers@lists.alioth.debian.org>
Date: Sat, 10 May 2014 22:52:23 +0200
Subject: Configure systemd service for Debian

* Regenerate dconf defaults from source files before start or reload
* Use the right executable name for Debian
* Delay restarts a little
* Don't install the display-manager.service alias, which is managed
  jointly by all Debian display managers via a debconf question

Authors: Josselin Mouette, Laurent Bigonville, Iain Lane
Last-Update: 2017-10-05
Origin: vendor, Debian
Forwarded: no
---
 data/gdm.service.in | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/data/gdm.service.in b/data/gdm.service.in
index 17e8a8d..066d763 100644
--- a/data/gdm.service.in
+++ b/data/gdm.service.in
@@ -20,14 +20,14 @@ After=rc-local.service plymouth-start.service systemd-user-sessions.service
 OnFailure=plymouth-quit.service
 
 [Service]
-ExecStart=${sbindir}/gdm
+ExecStartPre=/usr/share/gdm/generate-config
+ExecStart=${sbindir}/gdm3
 KillMode=mixed
 Restart=always
+RestartSec=1s
 IgnoreSIGPIPE=no
 BusName=org.gnome.DisplayManager
 EnvironmentFile=-${LANG_CONFIG_FILE}
+ExecReload=/usr/share/gdm/generate-config
 ExecReload=/bin/kill -SIGHUP $MAINPID
 KeyringMode=shared
-
-[Install]
-Alias=display-manager.service
